Detectors/TRD/workflow/src/TRDEventDisplayFeedSpec.cxx

```diff
@@ -0,0 +1,290 @@
+        trackJson["tlids"].push_back(trackletId);
+      }
+    }
+    trackArray.push_back(trackJson);
+  }
+  return trackArray;
+}
+
+json TRDEventDisplayFeedSpec::getTrackletsJson(gsl::span<const Tracklet64> tracklets, int iEvent)
+{
+  json trackletArray = json::array();
+
+  const auto& trigRec = mTrigRecs[iEvent];
+  for (int iTracklet = 0; iTracklet < trigRec.getNumberOfTracklets(); ++iTracklet) {
+    Tracklet64 tracklet = tracklets[iTracklet + trigRec.getFirstTracklet()];
+    CalibratedTracklet cTracklet = mTransformer.transformTracklet(tracklet);
+
+    std::string trackletId = Form("E%d_L%d", iEvent, iTracklet);
+    // Find matched track if it exists
+    std::string trackId = (mTrackletMap.find(trackletId) != mTrackletMap.end() ? mTrackletMap.at(trackletId) : "null");
+
+    int detector = tracklet.getDetector();
+    int sector = mGeo->getSector(detector);
+    int stack = mGeo->getStack(detector);
+    int layer = mGeo->getLayer(detector);
+
+    // Start position of both raw and calibrated tracklet in event display
+    float localY = cTracklet.getY();
+    // Slope of raw tracklet (key dyDxAN in JSON)
+    float rawDyDx = tracklet.getUncalibratedDy() / mGeo->cdrHght();
+    // Slope of calibrated tracklet
+    float dyDx = cTracklet.getDy() / mGeo->cdrHght();
+
+    json trackletJson = {
+      {"id", trackletId},
+      {"stk", stack},
+      {"sec", sector},
+      {"lyr", layer},
+      {"row", tracklet.getPadRow()},
+      {"trk", trackId},
+      {"lY", localY},
+      {"dyDx", dyDx},
+      {"dyDxAN", rawDyDx}};
+
+    if (trackId == "null") {
+      trackletJson["trk"] = nullptr;
+    }
+
+    trackletArray.push_back(trackletJson);
+  }
+  return trackletArray;
+}
+
+void TRDEventDisplayFeedSpec::writeDigits(gsl::span<const Digit> digits, int iEvent)
+{
+  const auto& trigRec = mTrigRecs[iEvent];
+  for (int det = 0; det < constants::MAXCHAMBER; det += 6) {
+    if (mUsedDetectors[det]) {
+      int sector = mGeo->getSector(det);
+      int stack = mGeo->getStack(det);
+
+      json digitsJson = {
+        {"evid", iEvent},
+        {"lyrs", json::array()}};
+
+      for (int iLayer = 0; iLayer < 6; iLayer++) {
+        json layerJson = {
+          {"lyr", iLayer},
+          {"pads", json::array()}};
+
+        for (int iDigit = trigRec.getFirstDigit(); iDigit < trigRec.getFirstDigit() + trigRec.getNumberOfDigits(); ++iDigit) {
+          Digit digit = digits[iDigit];
+
+          int detector = digit.getDetector();
+
+          if (detector == det + iLayer) {
+            // Digits are in stack with track
+            int row = digit.getPadRow();
+            int col = digit.getPadCol();
+
+            json padJson = {
+              {"row", row},
+              {"col", col},
+              {"tbins", json::array()}};
+
+            // Event display draws 24 time bins
+            int nTimebins = 25;
+            int t = 0;
+            for (auto adc : digit.getADC()) {
```

Comment:
Can the event display be changed to display 30 time bins as well? And since you do need a counter in this loop I would not use a range-based for loop, but an index one instead to improve the readability.

Reply:
Thanks Ole. I have changed the javascript code in the event display to show 30 timebins.

Reply:
Thanks for the PR @jbarrella . I've merged and extended so now a variable number of time-bins is accepted (but the hard-coded 20 timebins = 3cm drift remains).